tests/xtradbcluster/xtra_db_cluster_test.go

```diff
@@ -82,14 +91,58 @@ func TestXtraDBClusterAPI(t *testing.T) {
 
 	for _, cluster := range clusters.Clusters {
 		if cluster.Name == name {
-			assert.Equal(t, int32(3), cluster.Params.ClusterSize)
+			assert.Equal(t, int32(1), cluster.Params.ClusterSize)
 			assert.Equal(t, int64(1024*1024*1024), cluster.Params.Proxysql.ComputeResources.MemoryBytes)
 			assert.Equal(t, int32(1000), cluster.Params.Proxysql.ComputeResources.CpuM)
 			clusterFound = true
 		}
 	}
 	assert.True(t, clusterFound)
 
+	updateClusterReq := &controllerv1beta1.UpdateXtraDBClusterRequest{
+		KubeAuth: &controllerv1beta1.KubeAuth{
+			Kubeconfig: kubeconfig,
+		},
+		Name: name,
+		Params: &controllerv1beta1.XtraDBClusterParams{
+			ClusterSize: 2,
+			Pxc: &controllerv1beta1.XtraDBClusterParams_PXC{
+				ComputeResources: &controllerv1beta1.ComputeResources{},
+			},
+			Proxysql: &controllerv1beta1.XtraDBClusterParams_ProxySQL{
+				ComputeResources: &controllerv1beta1.ComputeResources{},
+			},
+		},
+	}
+
+	t.Log("Before first update")
+
+	updateXtraDBClusterResponse, err := tests.XtraDBClusterAPIClient.UpdateXtraDBCluster(tests.Context, updateClusterReq)
+	assert.NoError(t, err)
+	assert.NotNil(t, updateXtraDBClusterResponse)
+
+	t.Log("Waiting for cluster into changing state")
+	err = waitForClusterState(ctx, kubeconfig, controllerv1beta1.XtraDBClusterState_XTRA_DB_CLUSTER_STATE_CHANGING)
+	assert.NoError(t, err)
+
+	t.Log("Before second update")
+
+	// Cannot run a second update while the first haven't finish yet
+	updateXtraDBClusterResponse, err = tests.XtraDBClusterAPIClient.UpdateXtraDBCluster(tests.Context, updateClusterReq)
+	assert.Error(t, err)
+	assert.Nil(t, updateXtraDBClusterResponse)
+
+	err = waitForClusterState(ctx, kubeconfig, controllerv1beta1.XtraDBClusterState_XTRA_DB_CLUSTER_STATE_CHANGING)
```

Comment:
should we really wait for cluster state?

Reply:
Yes, otherwise the update fails because of the state checking.